composer.json

```diff
@@ -11,12 +11,13 @@
     }
   ],
   "require": {
-    "guzzlehttp/guzzle": "^7.0",
-    "php": "~8.1.0 || ~8.2.0 || ~8.3.0"
+    "guzzlehttp/guzzle": "^7.9.2",
+    "guzzlehttp/promises": "^2.0.4",
```

Comment:
I think you've just had to add the `guzzlehttp/promises` dependency to prevent the `--lowest` build dropping back to the 1.5.x series and failing on PHP8.4?

Please could you instead add a `require-dev` dependency for `guzzlehttp/promises >= 2.0.4`? That should achieve the same thing, but avoids us enforcing the constraint on end-users. And means if/when guzzle eventually release a version that supports promises@v3 our consumers and CI will be able to install that.